### PR TITLE
Add agreement hash to locking schemes

### PR DESCRIPTION
**Pull request by leviadam**

### CODE REVIEW

File: contracts/schemes/Auction4Reputation.sol [hunk 1]
```diff
@@ -80,6 +83,7 @@ contract Auction4Reputation {
         auctionReputationReward = _auctionReputationReward;
         reputationRewardLeft = _auctionReputationReward.mul(_numberOfAuctions);
         redeemEnableTime = _redeemEnableTime;
+        agreementHash = _agreementHash;
```
Comment (dkent600) [inline contracts/schemes/Auction4Reputation.sol]: Would it be helpful, for each contract, to require that the hash be != 0x0?

A 0x0 hash would not fulfill the purpose of the feature.

Comment (leviadam) [reply contracts/schemes/Auction4Reputation.sol]: I think it is good yo have 0, for the people who dont want this feature.

Comment (dkent600) [reply contracts/schemes/Auction4Reputation.sol]: OK, I have previous heard the opposite argument for not undertaking other suggestions I've made about the bootstrapping contracts:  That no one else is ever going to use these contracts.  lol

**Comment by orenyodfat** [comment]
how about adding AgreementHash contract  ?
 the locking contracts can inherit from this contract.
the contract will provide a modifier `onlyAgree()`

**Comment by leviadam** [comment]
LGTM

**Pull request merged**